pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java

```diff
@@ -501,4 +501,20 @@
      *            whether to auto update partition increasement
      */
     ConsumerBuilder<T> autoUpdatePartitions(boolean autoUpdate);
+
+    /**
+     * Set batch receive policy {@link BatchReceivePolicy} for consumer.
+     * By default, consumer will use {@link BatchReceivePolicy#DEFAULT_POLICY} as batch receive policy.
+     *
+     * Example:
+     * <pre>
+     * client.newConsumer().batchReceivePolicy(BatchReceivePolicy.builder()
+     *              .maxNumberOfMessages(100)
+     *              .maxSizeOfMessages(5 * 1024 * 1024)
+     *              .timeout(100)
```

Comment:
I think it is better to have a builder method `timeout(long timeout, TimeUnit timeoutUnit)` rather than having two separated methods.

Reply:
fixed